orangecontrib/prototypes/widgets/owimagegrid.py

```diff
@@ -0,0 +1,1141 @@
+import enum
+import itertools
+import logging
+import os
+from collections import namedtuple
+from concurrent.futures import Future
+from itertools import zip_longest
+
+import Orange.data
+import numpy as np
+from AnyQt.QtCore import (
+    Qt, QEvent, QSize, QSizeF, QRectF, QPointF, QUrl, QDir
+)
+from AnyQt.QtCore import pyqtSignal as Signal, pyqtSlot as Slot
+from AnyQt.QtGui import (
+    QPixmap, QPen, QBrush, QColor, QPainter, QImageReader
+)
+from AnyQt.QtWidgets import (
+    QGraphicsView, QGraphicsWidget, QGraphicsItem,
+    QGraphicsLinearLayout,
+    QGraphicsGridLayout, QSizePolicy, QApplication, QStyle, QShortcut,
+    QFormLayout)
+from Orange.widgets import widget, gui, settings
+from Orange.widgets.data.owimageviewer import ImageLoader, Preview, GraphicsScene
+from Orange.widgets.utils.itemmodels import VariableListModel
+from Orange.widgets.utils.overlay import proxydoc
+from Orange.widgets.widget import Input, Output
+
+from orangecontrib.prototypes.image_grid import ImageGrid
+
+_log = logging.getLogger(__name__)
+
+_ImageItem = namedtuple(
+    "_ImageItem",
+    ["index",  # Index in the input data table
+     "widget",  # GraphicsThumbnailWidget displaying the image.
+     "url",  # Composed final image url.
+     "future"]  # Future instance yielding an QImage
+)
+
+
+class OWImageGrid(widget.OWWidget):
+    name = "Image Grid"
+    description = "Visualize images in a similarity grid"
+    icon = "icons/Category-Prototypes.svg"
```

Comment:
Propose an icon.

Reply:
Let me know the next time you are in the lab and I'll show you how to propose an icon.

Reply:
Will do @ajdapretnar, thanks!